compiler/semstmts.nim

```diff
@@ -256,7 +273,7 @@ proc semTry(c: PContext, n: PNode; flags: TExprFlags): PNode =
   else:
     if n.lastSon.kind == nkFinally: discardCheck(c, n.lastSon.lastSon, flags)
     n.sons[0] = fitNode(c, typ, n.sons[0], n.sons[0].info)
-    for i in 1..last:
+    for i in 1..n.len-1:
```

Comment:
My fault, this should go from `1` to `n.len-1` if there's no finally block and to `n.len-2` otherwise (the `last` was needed after all)